nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -0,0 +1,366 @@
+import org.apache.nifi.annotation.behavior.SupportsBatching;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.flowfile.attributes.CoreAttributes;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.OutputStreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Set;
+import java.util.List;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+@PrimaryNodeOnly
+@SupportsBatching
+@InputRequirement(Requirement.INPUT_FORBIDDEN)
+@Tags({"twitter", "tweets", "social media", "status", "json"})
+@CapabilityDescription("Streams tweets from Twitter's streaming API v2. The stream provides a sample stream or a search "
+    + "stream based on previously uploaded rules. This processor also provides a pass through for certain fields of the "
+    + "tweet to be returned as part of the response. See "
+    + "https://developer.twitter.com/en/docs/twitter-api/data-dictionary/introduction for more information regarding the "
+    + "Tweet object model. \n\n"
+    + "Warning: the underlying Java SDK used is still in beta as of the publishing of this processor feature.")
+@WritesAttribute(attribute = "mime.type", description = "Sets mime type to application/json")
+public class ConsumeTwitter extends AbstractProcessor {
+
+    static final AllowableValue ENDPOINT_SAMPLE = new AllowableValue(StreamEndpoint.SAMPLE_ENDPOINT.getEndpointName(),
+            "Sample Stream",
+            "Streams about one percent of all Tweets. " +
+                    "https://developer.twitter.com/en/docs/twitter-api/tweets/volume-streams/api-reference/get-tweets-sample-stream");
+    static final AllowableValue ENDPOINT_SEARCH = new AllowableValue(StreamEndpoint.SEARCH_ENDPOINT.getEndpointName(),
+            "Search Stream",
+            "The endpoint that provides a stream of tweets that matches the rules you added to the stream. " +
+                    "If rules are not configured, then the stream will be empty. " +
```

Comment:
Recommend minor adjustments:
```suggestion
            "The search stream produces Tweets that match filtering rules configured on Twitter services. At least one filtering rule must be configured." +
```

Reply:
Making the changes and changing "one filtering rule" to "one well-formed filtering rule"